detection_rules/rule.py

```diff
@@ -515,6 +515,55 @@ def name(self) -> str:
     def type(self) -> str:
         return self.data.type
 
+    def _post_dict_transform(self, obj: dict) -> dict:
+        """Derive required fields from query AST."""
+
+        super()._post_dict_transform(obj)
+
+        self.add_related_integrations(obj)
+        self.add_required_fields(obj)
+        self.add_setup(obj)
+
```

Comment:
```suggestion

            # validate new field against the schema
            rule_type = obj['type']
            subclass =  self.get_data_subclass(rule_type)
            subclass.from_dict(obj)
```

Reply:
the `post_dict_transform` is called in additional places that modify the obj (e.g. adding `version`) which breaks validation. I've reorder the version obj insert to occur after validation to fit this approach.